apps/system/utils/utils_heapinfo.c

```diff
@@ -269,7 +261,22 @@ int utils_heapinfo(int argc, char **args)
 #if CONFIG_MM_NHEAPS > 1
 	heapinfo_init_totalinfo();
 #endif
-	struct mm_heap_s *heap = g_mmheap;
+#ifdef CONFIG_MM_KERNEL_HEAP
+	struct mm_heap_s *heap = g_kmmheap;
+#else
+	void *temp_addr = malloc(1);
+	if (temp_addr == NULL) {
+		printf("Fail to run heapinfo : Out of memory.\n");
+		return ERROR;
+	}
+	struct mm_heap_s *heap = mm_get_heap(temp_addr);
+	if (heap == NULL) {
+		printf("Fail to run heapinfo : Operation fail.\n");
```

Comment:
Would temp_addr be freed in mm_get_heap() when heap is null?

Reply:
Done. Thanks.